core/backend/src/BriefcaseManager.ts

```diff
@@ -573,6 +573,11 @@ export class BriefcaseManager {
     changesetProps.size = IModelJsFs.lstatSync(changesetProps.pathname)!.size;
 
     try {
+      // Refresh the access token if necessary
+      // Note: pushChanges could be a long running operations and should never be called through RPC, and we can use IModelHost's authorization client to fetch the access token
+      if (requestContext.accessToken.isExpired(5 * 60) && IModelHost.authorizationClient !== undefined) {

```

Comment:
I don't see how this is going to do anything unless the access token is within the safety expiry period. We don't have a way to refresh the token before that. If that's the case, this whole thing should just be:
```ts
requestContext.accessToken = await IModelHost.getAccessToken();
```

Reply:
Fixed.

Reply:
i simplified